### PR TITLE
fix: show errors

### PR DESCRIPTION
**Pull request by cmcartaya**

### CODE REVIEW

File: src/redux/actions/app/show-error-message.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
+export const SHOW_ERROR_MESSAGE = 'SHOW_ERROR_MESSAGE';
+export default function showErrorMessage(error) {
+    return {
+        type: SHOW_ERROR_MESSAGE,
+        message: error.message ? error.message : String(error),
```
Comment (LeandroTorresSicilia) [inline src/redux/actions/app/show-error-message.js]: you can use:
```error.message || String(error)```

File: src/redux/actions/authentication/index.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ import loginWithFacebook, {
     AUTH_ERROR_WITH_FACEBOOK,
 } from './facebook-login';
 import loginWithEmailAndPassword, { AUTH_START } from './email-password-login';
-import { AUTH_SUCCESS, AUTH_ERROR } from './handle-email-password-sign-in';
+import { AUTH_SUCCESS, SHOW_ERROR_MESSAGE } from './handle-email-password-sign-in';
```
Comment (LeandroTorresSicilia) [inline src/redux/actions/authentication/index.js]: SHOW_ERROR_MESSAGE is meant to be exported only in show-error-message.js and in the app/index.js, not here.

File: src/redux/actions/forgotPassword/index.js [hunk 3]
```diff
@@ -2,7 +2,7 @@ import resetForm, { RESET_FORM } from './reset-form';
 import sendEmailResetPassword, {
     SEND_PASS_RESET_EMAIL,
     RESET_EMAIL_SENT,
-    PASS_RESET_ERROR,
+    SHOW_ERROR_MESSAGE,
```
Comment (LeandroTorresSicilia) [inline src/redux/actions/forgotPassword/index.js]: same ^^

File: src/redux/actions/forgotPassword/reset-password.js [hunk 4]
```diff
@@ -2,7 +2,7 @@ import { sendPasswordResetEmail } from '../../services/firebase';
 
 export const SEND_PASS_RESET_EMAIL = 'SEND_PASS_RESET_EMAIL';
 export const RESET_EMAIL_SENT = 'RESET_EMAIL_SENT';
-export const PASS_RESET_ERROR = 'PASS_RESET_ERROR';
+export const SHOW_ERROR_MESSAGE = 'SHOW_ERROR_MESSAGE';
```
Comment (LeandroTorresSicilia) [inline src/redux/actions/forgotPassword/reset-password.js]: same ^^

File: src/redux/actions/registration/create-account.js [hunk 5]
```diff
@@ -5,7 +5,7 @@ import updateUser from '../../services/registration/update-user';
 
 export const CREATE_ACCOUNT_LOADING = 'CREATE_ACCOUNT_LOADING';
 export const CREATE_ACCOUNT_SUCCESS = 'CREATE_ACCOUNT_SUCCESS';
-export const CREATE_ACCOUNT_ERROR = 'CREATE_ACCOUNT_ERROR';
+export const SHOW_ERROR_MESSAGE = 'CREATE_ACCOUNT_ERROR';
```
Comment (LeandroTorresSicilia) [inline src/redux/actions/registration/create-account.js]: same ^^

File: src/redux/actions/registration/index.js [hunk 6]
```diff
@@ -1,15 +1,15 @@
 import createAccount, {
     CREATE_ACCOUNT_LOADING,
     CREATE_ACCOUNT_SUCCESS,
-    CREATE_ACCOUNT_ERROR,
+    SHOW_ERROR_MESSAGE,
```
Comment (LeandroTorresSicilia) [inline src/redux/actions/registration/index.js]: same

File: src/redux/reducers/app/index.js [hunk 7]
```diff
@@ -35,11 +21,11 @@ export default function (state = initialState, action) {
         case DONE_APP_INITIALIZATION:
             return state.set('isInitializing', false);
 
-        case START_APP_LOADING:
-            return startAppLoading(state, action.title, action.message);
+        case SHOW_ERROR_MESSAGE:
+            return state.set('errorMessage', action.error.message);
```
Comment (LeandroTorresSicilia) [inline src/redux/reducers/app/index.js]: Let's try to not set state directly here instead use a function like startAppLoading.
Also in the action you say:
```
return {
        type: SHOW_ERROR_MESSAGE,
        message: error.message ? error.message : String(error),
}
```
then here must be:
```state.set('errorMessage', action.message);```
and not ```action.error.message```

File: src/redux/reducers/authentication/index.js [hunk 8]
```diff
@@ -12,6 +11,7 @@ import {
     AUTH_SUCCESS_WITH_FACEBOOK,
     AUTH_ERROR_WITH_FACEBOOK,
 } from '../../actions/authentication';
+import { SHOW_ERROR_MESSAGE } from '../../actions/app/show-error-message';
```
Comment (LeandroTorresSicilia) [inline src/redux/reducers/authentication/index.js]: not, there is already a reducer listen for this action why do you need set it in others reducers if you only show the error based on app reducer.

File: src/redux/reducers/authentication/index.js [hunk 9]
```diff
@@ -68,7 +68,7 @@ export default function (state = initialState, action) {
         case AUTH_START:
             return authStart(state);
 
-        case AUTH_ERROR:
+        case SHOW_ERROR_MESSAGE:
             return authError(state, action.error);
```
Comment (LeandroTorresSicilia) [inline src/redux/reducers/authentication/index.js]: remove this case

File: src/redux/reducers/forgotPassword/index.js [hunk 10]
```diff
@@ -2,9 +2,9 @@ import { fromJS } from 'immutable';
 import {
     RESET_EMAIL_SENT,
     SEND_PASS_RESET_EMAIL,
-    PASS_RESET_ERROR,
     RESET_FORM,
 } from '../../actions/forgotPassword';
+import { SHOW_ERROR_MESSAGE } from '../../actions/app/show-error-message';
```
Comment (LeandroTorresSicilia) [inline src/redux/reducers/forgotPassword/index.js]: same ^^

File: src/redux/reducers/forgotPassword/index.js [hunk 11]
```diff
@@ -20,7 +20,7 @@ export default function (state = initialState, action) {
         case RESET_EMAIL_SENT:
             state = state.set('isEmailSent', true);
             return state.set('isLoading', false);
-        case PASS_RESET_ERROR:
+        case SHOW_ERROR_MESSAGE:
             state = state.set('error', action.error);
             return state.set('isLoading', false);
```
Comment (LeandroTorresSicilia) [inline src/redux/reducers/forgotPassword/index.js]: remove this case

File: src/redux/reducers/registration/index.js [hunk 12]
```diff
@@ -1,11 +1,11 @@
 /* eslint-disable no-param-reassign */
 import { fromJS } from 'immutable';
 import {
-    CREATE_ACCOUNT_ERROR,
     CREATE_ACCOUNT_LOADING,
     CREATE_ACCOUNT_SUCCESS,
     CREATE_ACCOUNT_RESET_FORM,
 } from '../../actions/registration';
+import { SHOW_ERROR_MESSAGE } from '../../actions/app/show-error-message';
```
Comment (LeandroTorresSicilia) [inline src/redux/reducers/registration/index.js]: same ^^

File: src/redux/reducers/registration/index.js [hunk 13]
```diff
@@ -19,7 +19,7 @@ export default function (state = initialState, action) {
             state = state.set('errorMessage', '');
             return state.set('isLoading', true);
 
-        case CREATE_ACCOUNT_ERROR:
+        case SHOW_ERROR_MESSAGE:
             /* eslint-disable no-param-reassign */
             state = state.set('isLoading', false);
             return state.set('errorMessage', action.error);
```
Comment (LeandroTorresSicilia) [inline src/redux/reducers/registration/index.js]: remove this case

**Comment by cmcartaya** [comment]
fix: #18

**Pull request merged**